### ISSUE TITLE
Fine-Tuning GR00T-N1 on Libero Spatial: Unexpected Impact of Denoise Steps on Performance

### ISSUE THREAD

**Issue by Rick0423** [NVIDIA/Isaac-GR00T#191]
First of all, thank you very much for open‐sourcing this incredible project—it has been extremely valuable to the community.

I have been experimenting with fine‐tuning the model on Libero_Spatial dataset, and then benchmarking it in the LIBERO simulation environment. During my tests, I noticed that as I increased the number of denoise steps , the overall accuracy actually decreased rather than improved.

I suspect that this behavior arises because, during fine‐tuning, each forward pass through the model only uses a single DiT pass (i.e., effectively a denoise step of 1). As a result, the distribution of the weights after fine‐tuning is biased toward operating with exactly one denoise iteration. However, at benchmark time, when the policy’s get_action(...) method is called, the model actually applies multiple DiT passes equal to the chosen denoise step count. This mismatch in distributions means that if you choose a denoise step closer to 1 (which matches the fine‐tuning condition), performance is better.

How should I fine‐tune more effectively?  
How should I select the most effective denoising_steps?

Thank you  so much for any suggestions！

**Comment by LukeLIN-web** [comment]
libero using single arm, how did you use humanoid robot on it ?
I find one issue may help: https://github.com/NVIDIA/Isaac-GR00T/issues/136#issuecomment-2833343460

**Comment by xwjiang2010** [comment]
> each forward pass through the model only uses a single DiT pass (i.e., effectively a denoise step of 1)

I don't think this statement is true. Each forward pass through the model does not literally correspond to a single DiT pass. Instead, we sample a timestep 𝑡∼𝑈(1,𝑇], add the corresponding level of noise to the input in one shot (using the closed-form formula for the noisy state), and then run the model once to predict the denoised target at that timestep.

Also we have just published the recipes to finetune on Libero datasets. PTAL https://github.com/NVIDIA/Isaac-GR00T/pull/357. 

Feel free to reopen if more issues. Thanks!
